rustsat/src/types.rs

```diff
@@ -697,6 +704,56 @@ impl Assignment {
             Some(var![self.assignment.len() as u32 - 1])
         }
     }
+
+    /// Read a solution from a SAT solver's output given the path
+    pub fn from_solver_output_path<P: AsRef<Path>>(path: P) -> anyhow::Result<SolverOutput> {
+        let reader = std::io::BufReader::new(
+            instances::fio::open_compressed_uncompressed_read(path)
+                .context("failed to open reader")?,
+        );
+        instances::fio::parse_sat_solver_output(reader)
+    }
+
+    /// Given a value line of a SAT solver's solution create an Assignment with
+    /// this line's literal values already included
+    pub fn from_vline(line: &str) -> anyhow::Result<Self> {
+        let mut assignment = Assignment::default();
+        assignment.extend_from_vline(line)?;
+        Ok(assignment)
+    }
+
+    /// Parses and saves literals from a value line.
+    pub fn extend_from_vline(&mut self, line: &str) -> anyhow::Result<()> {
+        if !line.starts_with("v ") {
```

Comment:
```suggestion
        for line in line.lines() {
            if !line.starts_with("v ") {
```
I just realized that a user could call this function with a huge string slice that already contains multiple v lines. Even though I would not recommend this, I think it might make sense to still support this usecase. (Of course close the for loop at the end and indent appropriately.)